src/library_sockfs.js

```diff
@@ -46,6 +46,7 @@ mergeInto(LibraryManager.library, {
       return FS.createNode(null, '/', {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */, 0);
     },
     createSocket: function(family, type, protocol) {
+      type &= ~{{{ cDefine('SOCK_CLOEXEC') }}}; // Some applications may pass it; it makes no sense for a single process.
```

Comment:
Perhaps add `|  cDefine('SOCK_CLOEXEC')` which is the other extra bit that can be set in the type of a linux socket.

Reply:
You mean SOCK_NONBLOCK  ?